packages/components/src/color-picker/alpha.js

```diff
@@ -34,34 +34,64 @@ import { noop } from 'lodash';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component, createRef } from '@wordpress/element';
+import { useState, useEffect, useRef } from '@wordpress/element';
 import { TAB } from '@wordpress/keycodes';
-import { pure } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import { calculateAlphaChange } from './utils';
 import KeyboardShortcuts from '../keyboard-shortcuts';
 
-export class Alpha extends Component {
-	constructor() {
-		super( ...arguments );
+export default function Alpha( { rgb, hsl, onChange = noop } ) {
+	const [ mouseDown, setMouseDown ] = useState( false );
+	const container = useRef();
 
-		this.container = createRef();
-		this.increase = this.increase.bind( this );
-		this.decrease = this.decrease.bind( this );
-		this.handleChange = this.handleChange.bind( this );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.handleMouseUp = this.handleMouseUp.bind( this );
-	}
+	const rgbString = `${ rgb.r },${ rgb.g },${ rgb.b }`;
+	const gradient = {
+		background: `linear-gradient(to right, rgba(${ rgbString }, 0) 0%, rgba(${ rgbString }, 1) 100%)`,
+	};
+	const pointerLocation = { left: `${ rgb.a * 100 }%` };
 
-	componentWillUnmount() {
-		this.unbindEventListeners();
-	}
+	const handleChange = ( e ) => {
+		const change = calculateAlphaChange( e, { hsl }, container.current );
+		if ( change ) {
+			onChange( change, e );
+		}
+	};
+
+	const handleMouseDown = () => {
+		window.addEventListener( 'mousemove', handleChange );
+		window.addEventListener( 'mouseup', handleMouseUp );
+	};
 
-	increase( amount = 0.01 ) {
-		const { hsl, onChange = noop } = this.props;
+	const handleMouseUp = () => {
+		setMouseDown( false );
+	};
+
+	const unbindEventListeners = () => {
+		window.removeEventListener( 'mousemove', handleChange );
```

Comment:
In #26810, I'm about to warn against using globals for this. Might be nice to try to use a ref here.

Reply:
I used `ownerDocument` instead of `window` object in daddou-ma/gutenberg@490bb10

Reply:
some unit tests are failing after using `ownerDocument`, i will check what's the problem.

Reply:
@ellatrix i think `react-test-renderer` has a problem dealing with `refs`, `container.current` is `null` when running tests. do ou have an idea why ?

failed unit test : `packages/components/src/color-picker/test/index.js`

Reply:
Thanks, i will try it

Reply:
Thank you, it solved the problem of `refs`, i added `createNodeMock` to `ColorPicker` tests in daddou-ma/gutenberg@fe82dbc